grpc-client-server/prober_client/main_test.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	pb "github.com/Berkeli/immersive-go/grpc-client-server/prober"
+	"github.com/acarl005/stripansi"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+type mockProbeServer struct {
+	pb.UnimplementedProberServer
+}
+
+func (*mockProbeServer) DoProbes(ctx context.Context, req *pb.ProbeRequest) (*pb.ProbeReply, error) {
+	if req.Endpoint == "http://musterror:8080" {
+		return nil, grpc.Errorf(15, "error")
+	}
+	return &pb.ProbeReply{
+		AverageResponseTime: 123.123,
+		FailedRequests:      0,
+	}, nil
+}
+
+func dialer() func(context.Context, string) (net.Conn, error) {
+	listener := bufconn.Listen(1024 * 1024)
+
+	server := grpc.NewServer()
+
+	pb.RegisterProberServer(server, &mockProbeServer{})
+
+	go func() {
+		if err := server.Serve(listener); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	return func(context.Context, string) (net.Conn, error) {
+		return listener.Dial()
+	}
+}
+
+var lis = bufconn.Listen(1024 * 1024)
+
+func TestArrayFlag(t *testing.T) {
+	t.Run("String", func(t *testing.T) {
+		tests := map[string]struct {
+			i    *ArrayFlag
+			want string
+		}{
+			"empty": {
+				i:    &ArrayFlag{},
+				want: "",
+			},
+			"1 element": {
+				i:    &ArrayFlag{"foo"},
+				want: "foo",
+			},
+			"2 elements": {
+				i:    &ArrayFlag{"foo", "bar"},
+				want: "foo, bar",
+			},
+		}
+		for name, tt := range tests {
+			t.Run(name, func(t *testing.T) {
+				require.Equal(t, tt.want, tt.i.String())
+			})
+		}
+	})
+
+	t.Run("Set", func(t *testing.T) {
+		i := &ArrayFlag{}
+		tests := map[string]struct {
+			value string
+			want  *ArrayFlag
+		}{
+			"add element": {
+				value: "foo",
+				want:  &ArrayFlag{"foo"},
+			},
+			"add another element": {
```

Comment:
I'd say "add two elements" rather than "add another element" - another implies it's related to something else that already happened (e.g. that your two test cases depend on each other, which they don't)

Reply:
It's actually adding another element, there's no option to add multiple at once

Reply:
The difference I'm trying to highlight is that if you have two test cases: "add element" and "add another element", they read as if possibly related.

Can you think of a name which _doesn't_ suggest that?